drivers/flash/nrf_qspi_nor.c

```diff
@@ -536,19 +536,38 @@ static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 static int qspi_nor_write(struct device *dev, off_t addr, const void *src,
 			  size_t size)
 {
+	const void *sptr = src;
+	size_t dlen = size;
+	u8_t buf[4];
+
 	if (!src) {
 		return -EINVAL;
 	}
 
-	/* write size must be non-zero multiple of 4 bytes */
-	if (((size % 4U) != 0) || (size == 0)) {
-		return -EINVAL;
-	}
 	/* address must be 4-byte aligned */
 	if ((addr % 4U) != 0) {
 		return -EINVAL;
 	}
 
+	/* Since the QSPI driver requires data to be at least 4 bytes we need
+	 * to use a 4 byte buffer for writes smaller than 4 bytes.
+	 */
+	if (size < 4U) {
```

Comment:
This also needs a `size > 0` check.

Reply:
Moved the size check above. Since `size` is `typeof size_t` it's safe to assume that `size` can't be less than 0 right.

Reply:
I think he has done it at line 543.